src/pages/preferences/preferencesTabs/cliTab/DownloadCard.tsx

```diff
@@ -1,68 +1,121 @@
-        title="Linux (64-bit)"
-        link={cliDownloadUrls.linux_amd64}
+export const DownloadCard = () => {
+  const { data, loading } = useQuery<
+    ClientConfigQuery,
+    ClientConfigQueryVariables
+  >(GET_CLIENT_CONFIG);
+
+  if (loading) {
+    return <Skeleton active paragraph={{ rows: 6 }} />;
+  }
+  const clientBinaries = get(data, "clientConfig.clientBinaries", []);
+  return (
+    <Container>
+      <Subtitle>Command-Line Client</Subtitle>
+      <CardDescription>
+        <Body>
+          View the{" "}
+          <StyledLink href={cliDocumentationUrl}>documentation</StyledLink> or
+          run{" "}
+        </Body>
+        <InlinePre>evergreen --help or evergreen [command] --help</InlinePre>{" "}
+        <Body>for additional assistance.</Body>
+      </CardDescription>
+      <CardGroup>
+        <CliDownloadBox
+          title="Linux (64-bit)"
+          link={getBinaryUrl("linux", "amd64", clientBinaries)}
+        />
+        <CliDownloadBox
+          title="MacOS"
+          link={getBinaryUrl("darwin", "amd64", clientBinaries)}
+        />
+        <CliDownloadBox
+          title="Windows"
+          link={getBinaryUrl("windows", "amd64", clientBinaries)}
+        />
+      </CardGroup>
+      <Accordian
+        title={<StyledLink>Show More</StyledLink>}
+        toggledTitle={<StyledLink>Show Less</StyledLink>}
+        contents={<ExpandableLinkContents clientBinaries={clientBinaries} />}
+        toggleFromBottom
+        showCaret={false}
       />
-      <CliDownloadBox title="MacOS" link={cliDownloadUrls.darwin_amd64} />
-      <CliDownloadBox title="Windows" link={cliDownloadUrls.windows_amd64} />
-    </CardGroup>
-    <Accordian
-      title={<StyledLink>Show More</StyledLink>}
-      toggledTitle={<StyledLink>Show Less</StyledLink>}
-      contents={<ExpandableLinkContents />}
-      toggleFromBottom
-      showCaret={false}
-    />
-  </Container>
-);
+    </Container>
+  );
+};
 
 interface CliDownloadBoxProps {
   title: string;
-  link: string;
+  link: string | null;
 }
 const CliDownloadBox: React.FC<CliDownloadBoxProps> = ({ title, link }) => (
   <CliDownloadCard>
     <CliDownloadTitle>{title}</CliDownloadTitle>
-    <CliDownloadButton href={link} as="a">
+    <CliDownloadButton href={link} disabled={link === null} as="a">
       Download
     </CliDownloadButton>
   </CliDownloadCard>
 );
 
-const ExpandableLinkContents = () => (
+interface ExpandableLinkContentsProps {
+  clientBinaries: ClientBinary[];
+}
+const ExpandableLinkContents: React.FC<ExpandableLinkContentsProps> = ({
+  clientBinaries,
+}) => (
   <LinkContainer>
-    <StyledLink href={cliDownloadUrls.linux_arm64}>Linux ARM 64-bit</StyledLink>
-    <StyledLink href={cliDownloadUrls.linux_s390x}>Linux zSeries</StyledLink>
-    <StyledLink href={cliDownloadUrls.linux_386}>Linux 32-bit</StyledLink>
-    <StyledLink href={cliDownloadUrls.linux_ppce64le}>
+    <StyledLink href={getBinaryUrl("linux", "arm64", clientBinaries)}>
```

Comment:
Why not map over the returned array for these components?

Reply:
Hey @john-m-liu  i did it this way intentionally. There are 3 elements that have their own prominent cards and the rest are displayed as links. So i would have to filter those elements out in order to iterate through the list. All the elements also have distinct display names that aren't present in the data. It seemed simpler to fetch this data manually instead of dealing with transforming the data so i can use it.

Reply:
You'll need to have logic to extract the display name in both approaches, but mapping over the returned array accounts for results that aren't listed here, either because we just added them or forgot to handle here. I think that approach will be better and more future-proof

Reply:
That makes sense just pushed up those changes.